domainservice/src/main/kotlin/com/makarytskyi/rentcar/config/KafkaConfiguration.kt

```diff
@@ -0,0 +1,77 @@
+package com.makarytskyi.rentcar.config
+
+import com.makarytskyi.internalapi.subject.KafkaTopic
+import org.apache.kafka.clients.consumer.ConsumerConfig
+import org.apache.kafka.clients.producer.ProducerConfig
+import org.apache.kafka.common.serialization.ByteArrayDeserializer
+import org.apache.kafka.common.serialization.ByteArraySerializer
+import org.apache.kafka.common.serialization.StringDeserializer
+import org.apache.kafka.common.serialization.StringSerializer
+import org.springframework.boot.autoconfigure.kafka.KafkaProperties
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+import reactor.kafka.receiver.KafkaReceiver
+import reactor.kafka.receiver.ReceiverOptions
+import reactor.kafka.sender.KafkaSender
+import reactor.kafka.sender.SenderOptions
+
+@Configuration
+class KafkaConfiguration(
+    private val kafkaProperties: KafkaProperties
+) {
+
+    @Bean
+    fun kafkaSender(): KafkaSender<String, ByteArray> {
+        return createProducer(producerOptions())
+    }
+
+    @Bean
+    fun createRepairingKafkaReceiver(): KafkaReceiver<String, ByteArray> {
+        return createReceiver(consumerOptions(), GROUP_ID, KafkaTopic.REPAIRING_CREATE)
+    }
+
+    private fun createReceiver(
+        receiverOptions: MutableMap<String, Any>,
+        groupId: String,
+        topic: String
+    ): KafkaReceiver<String, ByteArray> {
+        receiverOptions[ConsumerConfig.GROUP_ID_CONFIG] = groupId
+        val options = ReceiverOptions.create<String, ByteArray>(receiverOptions)
+            .subscription(setOf(topic))
+        return KafkaReceiver.create(options)
+    }
+
+    private fun createProducer(
+        producerOptions: MutableMap<String, Any>
+    ): KafkaSender<String, ByteArray> {
+        return KafkaSender.create(
+            SenderOptions.create(producerOptions)
+        )
+    }
+
+    private fun producerOptions(): MutableMap<String, Any> {
```

Comment:
why `MutableMap` is used?

Reply:
But here it is better to use Map instead, because I don't add to this map any options later. I will fix it.